tracks/python/exercises/square-root/mentoring.md

```diff
@@ -0,0 +1,76 @@
+# Mentoring
+
+## Problem and Challenges
+
+The problem asks us to find the square roots of numbers without using builtin functions or operators, like `**`, `pow`, `math.pow`, `math.sqrt` or `sum`.
```

Comment:
Why would `sum` not be allowed?

Reply:
I'm not sure exactly why, but it's mentioned in the exercise instructions as an example builtin to avoid.
> Python offers a wealth of mathematical functions in the form of the [math module](https://docs.python.org/3/library/math.html) and built-ins such as [pow()](https://docs.python.org/3/library/functions.html#pow) and [sum()](https://docs.python.org/3/library/functions.html#sum). However, we'd like you to consider the challenge of solving this exercise without those built-ins or modules.

I can see at least 2 commmunity solutions where this function is used, inappropriately I might say:
1. https://exercism.org/tracks/python/exercises/square-root/solutions/PetaGb
2. https://exercism.org/tracks/python/exercises/square-root/solutions/soroushmz

Reply:
Mentoring notes can't really force a student to change their solution and I don't think the first couple lines suggest it absolutely _shouldn't_ be done using `**`. Students could even hardcode values to fool the tests.

I imagine that people who asked for mentorship did so for 2 reasons:
1. Their solution is too simple and are wondering if there's more to it,
2. They tried implementing one of the methods in the wiki articles and want to improve on that implementation.

The spirit of the exercise is not to use a builtin operator, but to investigate other methods of doing the calculation without it.

Heron's method happens to be the simplest, its the first method linked in the [wiki page](https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Heron's_method), there's no intermediate step between `**` or `math.pow` and that.

It's Heron's method that's the intermediary step, Newton's method and binary search (included as an alternative solution) follow the same path of shrinking the lookup space.

I could add the simpler solution of:
```py
def square_root(number):
    for num in range(number+1):
        if num * num == number:
            return num
```
But this isn't really related to Heron's method, IMO they're distinct methods.

As for a `sum()` solution, I can't think of an appropriate one. The two `sum()` community solutions linked above sum just one number from the range. The snippet linked just above is arguably an improvement on it.